mlx_graphs/utils/convert.py

```diff
@@ -1,4 +1,8 @@
-from typing import Any
+from collections import defaultdict
+from typing import Any, Union
+
+import mlx.core as mx
+import networkx as nx
```

Comment:
this requires networkx as a main dependency (which I'm still not sure we want to have)

Reply:
Is there a way to not add it and still have `Union[nx.Graph]` as input? I tried to remove it but it keeps giving me the error related to dependency

Reply:
You can wrap the import in a `try/except`

```python
try:
    import networkx as nx
except ImportError:
    raise ImportError("networkx is required to convert from nextworkx graphs - run `pip install networkx`")
```

Reply:
I'll check whether that causes issues when compiling the documentation